### DISCUSSION TITLE
Save A Copy - Data Corruption

### DISCUSSION THREAD

**Discussion by st11x** [directus/directus#13516] [Archive]
When using the save a copy, the relational fields (M2A in this case) are not copied and in those fields in the new asset, they are shown as "Invalid Item". 

Further, when you delete the new asset, it corrupts the data in the join table. 

I see several bugs raised on this, but I did not see anything about the data corruption.

### To Reproduce

Create a collection with M2A fields. Create a new asset and duplicate it like this:

![image](https://user-images.githubusercontent.com/7398499/170093329-15ed3111-a702-4fd9-b1f4-f6985be90680.png)

In the new asset screen, all the relational data is invalid. In other cases, it never got to this page and the loader kept spinning. 

![image](https://user-images.githubusercontent.com/7398499/170094270-061db7f9-d3e0-45fc-bf8b-3438810f386c.png)

Inspecting the join table, we can see that it created the rows, but it didn't duplicate the columns. 

![{4638F541-891B-42AF-8945-9D8C70E86692}](https://user-images.githubusercontent.com/7398499/170094379-b6fd27bf-83f1-4e5e-a94b-83064267abbb.png)

And if you delete the new asset, it leaves the rows behind and nulled the asset id

![image](https://user-images.githubusercontent.com/7398499/170094637-5b8e36a8-02a0-4e1c-b8ef-05b4b4ebfe59.png)

**Comment by rijkvanzanten** [comment]
What's your configuration for Item Duplication Fields under the collection settings? We'll track any issues with relational fields during copy under https://github.com/directus/directus/issues/13406 👍🏻

**Reply by st11x** [reply]
I don't have anything set up for any collection in the project.

**Reply by st11x** [reply]
I found that the last screenshot where it sets the asset id to null instead of deleting the row, this happens on a regular deletion too. So it is not just a problem with saving a copy of the asset.
